### DISCUSSION TITLE
stuck while running prisma db push or pull

### DISCUSSION THREAD

**Discussion by BNK989** [prisma/prisma#24412] [Q&A] [answered]
### Question

I have a new nuxt project in which I'm trying to connect to my db in supabase, I've installed prisma and initiated it.
in the file prisma/schema I've added my test schema, and I'm trying to push it using 'pnpx prisma db push' I receive no error or anything like that and it seems like the connection was made (if i change the login link it throws an error. here the link I have - removed the password postgres://postgres.baqeisvusnvwlldjnlie:********@aws-0-eu-central-1.pooler.supabase.com:6543/postgres

I also tried going in to supabase, creating a table and then tried pnpx prisma db pull and have a similar result, no error just a running Introspecting based on datasource defined in prisma\schema.prisma
![image](https://github.com/prisma/prisma/assets/89899137/29c9990d-b3a5-44e0-bf90-853732fd960c)

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @BNK989 👋 

Thank you for raising this question.

Can you try to use the direct connection string rather than the pooled one. Here is an example of a direct connection string:
```
DIRECT_URL="postgres://postgres:__PASSWORD__@db.__YOUR_SUPABASE_PROJECT__.supabase.co:5432/postgres"
```
The default port for the pooler is 6543, but if you are using the direct connection, it should be 5432. Ensure that you are using the correct port in your connection string.

**Reply by BNK989** [reply]
Amazing I ended up just changing the port and it worked! thank you

**Reply by AideFlood** [reply]
Can confirm - I changed the port from 6543 to 5432 - it's working now!

**Reply by janpio** [reply]
Note that this is now also documented at Supabase: https://supabase.com/partners/integrations/prisma

**Reply by basitng** [reply]
> Can confirm - I changed the port from 6543 to 5432 - it's working now!

changing the port also worked for me

**Reply by NoraldenElhouni** [reply]
i change the port and it works thanks alot

**Reply by GoldeggCoder** [reply]
Also helped me! Thanks

**Reply by stpn48** [reply]
Bro thank you worked for me ❤️

**Reply by amineanime1** [reply]
Thank YOUU finally ❤️

**Reply by darren-finch** [reply]
Love when something is this simple, thanks man.

**Reply by Brien-Austin** [reply]
yeah the same worked for me after trying to figure out for a hour

**Comment by BNK989** [comment]
Amazing
Thank you,

Ben N. Klinovsky

**Comment by guiyumin** [comment]
Thanks. It worked!

**Comment by sreenivasanac** [comment]
thanks, it worked.
Can we also have it as a comment when someone is running the command, something like
"If this command is taking more time, try moving the port from 5432 instead of default port 6543"

**Comment by robcrock** [comment]
It's unfortunate that this was the solution... Prisma needs to update the backend to reflect what users on the frontend are presented with, because it was this port number.

**Comment by geforcesong** [comment]
stupid enough

**Comment by abdullah-azeemi** [comment]
Thanks
